jsk_hand/softhand_ros/README.md

```diff
@@ -0,0 +1,178 @@
+# softhand_ros
+
+ROS package for SoftHand
+
+## Note
+
+This package depends on the branch below.
+
+- [pazeshun/dynamixel_motor@gripper-v6-devel](https://github.com/pazeshun/dynamixel_motor/tree/gripper-v6-devel)
+
+## Installation
+
+### Workspace build
+
+```bash
+source /opt/ros/$ROS_DISTRO/setup.bash
+mkdir ~/softhand_ws/src -p
+cd ~/softhand_ws/src
+wget https://raw.githubusercontent.com/knorth55/softhand_ros/master/fc.rosinstall -O .rosinstall
+wstool up
+rosdep install --ignore-src --from-paths . -y -r -i
+cd ~/softhand_ws
+catkin build
+```
+
+### Udev installation
+
+```bash
+source ~/softhand_ws/devel/setup.bash
+roscd softhand_ros
+sudo cp udev/*.rules /etc/udev/rules.d
+sudo service udev reload
+sudo service udev restart
+```
+
+## How to use
+
+### Launch softhand
+
+#### For softhand v1
+
+```bash
+source ~/softhand_ws/devel/setup.bash
+# for left softhand v1
+roslaunch softhand_ros softhand_left.launch
+# for right softhand v1
+roslaunch softhand_ros softhand_right.launch
+```
+
+#### For softhand v2
+
+```bash
+source ~/softhand_ws/devel/setup.bash
+# for left softhand v2
+roslaunch softhand_ros softhand_v2_left.launch
+# for right softhand v2
+roslaunch softhand_ros softhand_v2_right.launch
+```
+
+### Control softhand by euslisp
+
+#### For softhand v1
+
+```bash
+source ~/softhand_ws/devel/setup.bash
+roscd softhand_ros/euslisp
+roseus softhand-interface.l
+# euslisp interactive mode
+# (softhand-init)
+# (send *ri* :start-grasp)
+# (send *ri* :stop-grasp)
+```
+
+#### For softhand v2
+
+```bash
+source ~/softhand_ws/devel/setup.bash
+roscd softhand_ros/euslisp
+roseus softhand-v2-interface.l
+# euslisp interactive mode
+# (softhand-v2-init)
+# (send *ri* :close-thumb)
+# (send *ri* :open-thumb)
+# (send *ri* :start-grasp)
+# (send *ri* :stop-grasp)
+```
+
+## Softhand hardware installation
+
+### Set baud rate
+
+#### For softhand v1
+
+```bash
+# set baud rate from 57600 to 1000000
+rosrun dynamixel_driver set_servo_config.py -b 57600 -r 1 MOTOR_ID
+```
+
+#### For softhand v2
+
+```bash
+# set baud rate from 57600 to 57143
+rosrun dynamixel_driver set_servo_config.py -b 57600 -r 34 MOTOR_ID
+```
+
+### Set motor ID
+
+```bash
+rosrun dynamixel_driver change_id.py OLD_MOTOR_ID NEW_MOTOR_ID
+```
+
+#### Motor IDs of Softhand v1
+
+- 1: Thumb
+- 2: Index finger
+- 3: Middle finger
+
+#### Motor IDs of Softhand v2
+
+- 1: Thumb rotate
+- 2: Thumb
+- 3: Index finger
+- 4: Middle finger
+
+### Disable overload error
+
+#### For softhand v1
+
+```python
+import roslib
+roslib.load_manifest('dynamixel_driver')
```

Comment:
we do not need `import roslib, roslib.load_manifest` https://answers.ros.org/question/107261/roslibload_manifest-error-python/?answer=107561#post-id-107561

Reply:
@pazeshun do we need this `roslib`?
I want to make sure.
It seems not necessary for me, too.

Reply:
I think I just copied these lines from dynamixel_motor, which seems very old:
https://github.com/arebgun/dynamixel_motor/blob/3d842328641a8c62f90388a254e5efb12929c1ae/dynamixel_driver/src/dynamixel_driver/dynamixel_serial_proxy.py#L52-L53
So perhaps it is not necessary.
I'll check with real dynamixel motor.